packages/cpp/ArmoniK.Api.Client/source/submitter/ResultsClient.cpp

```diff
@@ -64,21 +66,23 @@ void ResultsClient::upload_result_data(const std::string &session_id, const std:
 
   while (offset < payload.size()) {
     size_t chunkSize = std::min(maxChunkSize, payload.size() - offset);

```

Comment:
you could simplify this like this:
```suggestion
  while (payload.size()) {
    auto chunk = payload.substr(0, maxChunkSize);
    payload = payload.substr(maxChunkSize);
```